src/components/Popover/index.tsx

```diff
@@ -117,17 +122,36 @@ export default function Popover({
   const updateCallback = useCallback(() => {
     update && update()
   }, [update])
-  useInterval(updateCallback, show ? 100 : null)
+  useInterval(updateCallback, show ? 200 : null)
+
+  useEffect(() => {
+    if (show) {
+      if (!delayShowTimeout) {
+        setShouldShow(true)
+        return
+      }
+      const tooltipTimer = setTimeout(() => {
+        setShouldShow(true)
+      }, delayShowTimeout)
+
+      return () => {
+        clearTimeout(tooltipTimer)
+      }
+    } else {
+      setShouldShow(false)
+      return
+    }
+  }, [show, delayShowTimeout])
 
   return showInline ? (
-    <PopoverContainer show={show}>{content}</PopoverContainer>
+    <PopoverContainer show={shouldShow}>{content}</PopoverContainer>
```

Comment:
we no longer need to do `show && shouldShow` here now, shouldShow is the single source of truth for state here